src/main/java/seedu/duke/command/ClassificationCommand.java

```diff
@@ -0,0 +1,65 @@
+package seedu.duke.command;
+
+import seedu.duke.exception.DukeException;
+
+import java.util.HashMap;
+
+public abstract class ClassificationCommand extends Command {
+    private static final String ADD_COMMAND = "add";
+    private static final String REMOVE_COMMAND = "remove";
+    private static final String UNCATEGORISED = "uncategorised";
+    private static final String UNCATEGORISED_ERROR = "As 'Uncategorised' is a default classification, you are unable" +
+            "to delete it.";
+
+    public ClassificationCommand(String description, HashMap<String, String> args) {
+        super(description, args);
+    }
+
+    protected String validateInput() throws DukeException {
+        String invalidInput = "";
+        if (getClass() == CategoryCommand.class) {
+            invalidInput = "Your category input is empty/invalid :(";
+        } else if (getClass() == GoalCommand.class) {
+            invalidInput = "Your goal input is empty/invalid :(";
+        }
+
+        if (getArgs().isEmpty()) {
+            throw new DukeException(invalidInput);
+        } else if (getArgs().containsKey(ADD_COMMAND) && getArgs().containsKey(REMOVE_COMMAND)) {
+            errorMessage(invalidInput);
+        }
+        String arg;
+        if (getArgs().containsKey(ADD_COMMAND)) {
+            arg = getArg(ADD_COMMAND);
+            checkArg(arg, invalidInput);
```

Comment:
If we pass "uncategorised" as value to the add parameter, the error should no longer be UNCATEGORISED_ERROR - "to delete it"